### PR TITLE
Update @redhat-developer/vscode-redhat-telemetry to 0.2.0

### LINKED ISSUE #365: Update @redhat-developer/vscode-redhat-telemetry to 0.1.1
This removes the dependency to vscode-commons.

See https://github.com/redhat-developer/vscode-redhat-telemetry#add-the-below-code-to-your-extensionts for instructions on how to implement this change.

~~Note that `redhat.telemetry.enabled` doesn't need to be added to the `package.json`, since it will be contributed by vscode-java~~ see note from Fred

### PR DESCRIPTION
**Pull request by datho7561**
Closes #365

Signed-off-by: David Thompson <davthomp@redhat.com>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -147,6 +146,12 @@
     "configuration": {
       "title": "Quarkus Tools",
       "properties": {
+        "redhat.telemetry.enabled": {
```
Comment (rgrunber) [inline package.json]: According to https://github.com/redhat-developer/vscode-redhat-telemetry#contribute-the-redhattelemetryenabled-preference , `Unless your extension already depends on a telemetry-enabled Red Hat extension, it needs to declare the redhat.telemetry.enabled preference in its package.json, like:` . So does this need to be here ? This depends on vscode-microprofile, which defines this property.

**Review by rgrunber — CHANGES_REQUESTED** [review]
I do seem to be getting the warning about vscode-quarkus being unable to register redhat.telemetry.enabled because it's already registered.

Otherwise, startup/shutdown/commands seem to be reported correctly.

**Comment by fbricon** [comment]
No, redhat.telemetry.enabled shouldn't be defined here, as per the doc

Le ven. 23 juil. 2021 à 01:34, Roland Grunberg ***@***.***> a
écrit :

File: src/utils/telemetryUtils.ts [hunk 2]
```diff
@@ -30,10 +44,14 @@ export async function sendCommandFailedTelemetry(commandName: string, msg?: stri
  *
  * @param commandName the name of the command that was run
  * @param suffix the suffix to add to the command to get the event name
+ * @throws if the telemetry service has not been initialized yet
+ * @returns when the telemetry
```
Comment (rgrunber) [inline src/utils/telemetryUtils.ts]: "when the telemetry event has been sent"

**Review by rgrunber — APPROVED** [review]
Change seems to be working for me. Tested on a vsix.

**Comment by datho7561** [comment]
Thanks for the review, Roland!

**Pull request merged**